scripts/maint/rename_c_identifier.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/python3
+
+"""
+Helpful script to replace one or more C identifiers, and optionally
+generate a commit message explaining what happened.
+"""
+
+import argparse
+import fileinput
+import os
+import re
+import shlex
+import subprocess
+import sys
+import tempfile
+
+TOPDIR="src"
+
+def is_c_file(fn):
+    """
+       Return true iff fn is the name of a C file.
+
+       >>> is_c_file("a/b/module.c")
+       True
+       >>> is_c_file("a/b/module.h")
+       True
+       >>> is_c_file("a/b/module.c~")
+       False
+       >>> is_c_file("a/b/.module.c")
+       False
+       >>> is_c_file("a/b/module.cpp")
+       False
+    """
+    fn = os.path.split(fn)[1]
+    if fn.startswith("."):
+        return False
+    ext = os.path.splitext(fn)[1]
+    return ext in {".c",".h",".i","inc"}
+
+def list_c_files(topdir=TOPDIR):
+    """
+       Use git to list all the C files under version control.
+
+       >>> lst = list(list_c_files())
+       >>> "src/core/mainloop/mainloop.c" in lst
+       True
+       >>> "src/core/mainloop/twiddledeedoo.c" in lst
+       False
```

Comment:
Can you add an example for a generated C file?
We generate the git version include file.

Reply:
f7cde07da5a0b554406ba0c08c6ebbe7957515db